src/hooks/sources/useDeleteSource.tsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from 'react';
+import { SourceWithId } from '../../interfaces/Source';
+
+export function useDeleteSource(source: SourceWithId | null) {
```

Comment:
I think we could simplify this hook so that instead of deleting as an effect we just delete when called i.e 

const  [deleteSource, loading] = useDeleteSource();

...

<EditView
              source={currentSource}
              updateSource={(source) => setUpdatedSource(source)}
              close={() => setCurrentSource(null)}
              deleteInventorySource={(source) => deleteSource(source)}
            />